### ISSUE TITLE
Suggestion: Drone base

### ISSUE THREAD

**Issue by Ahammer18** [tgstation/-tg-station#7762]
Drones having their own base that is connected to the station via vents would be really cool. This would cause several positive things, such as:

-Drone no longer build autism forts
-quicker response time if they have a central hub
-No more stealing tools (load it with an autolathe)
-They'd stay out of people's way
-If you include some fun things, they might be less tempted to leave on the shuttle

This could go in several places:

-Engineering construction site
-Science testing lab (rarely used)
-New 'drone' satellite
-Part of the AI satellite
-Arrivals Vacant Office
-New room added to the side of the station or something

**Comment by TheVekter** [comment]
I am 110% for this. Y'know what works great for this? The electrical maint north of the Chapel, it's not used for jack shit right now.

**Comment by Lumipharon** [comment]
I dont really see this as neccessary, considering they're an entirely optional addition to the station.
Drones also already spawnwith a fully loaded toolbox, unless something has changed recently.

This is just changing drones 'building autism forts' to drone 'starting with an autism fort'.

Drones are just a little something for dead people to preoccupy themselves with when the round isn't ending.

**Comment by paprka** [comment]
I added a drone fabricator in another PR so the room that spawns in will kind of act like the drone base. It's on the construction area (above tcomms) by default.

**Comment by Ahammer18** [comment]
Lumipharon, this would be taking up an unused area of the station. 

http://puu.sh/fYquP/82085d2892.png

This is what I came up with, forgive my horrible mapping, and the wonky pipes

**Comment by paprka** [comment]
I like the idea of there being an autolathe and a fabricator there. You can remap this stuff once my PR has been accepted since it literally just tosses a drone fab down in the middle of that room.

**Comment by TheVekter** [comment]
Is that an exofab or a drone fabricator? Dronefab needs a new sprite to
differentiate itself.

**Comment by paprka** [comment]
It has a unique sprite, yeah, but my PR hasn't been accepted. It looks like this:

![adfaf](https://cloud.githubusercontent.com/assets/7247232/6206321/0a5217aa-b53c-11e4-98a5-13037756648b.gif)

**Comment by Steelpoint** [comment]
The room itself should not be too big, just enough to fit the fabricator and some spare equipment.

**Comment by TheVekter** [comment]
It needs to have metal and glass spawn in there, and preferably (if no one
has any complaints) be inaccessible to the crew. Make it where there's only
a vent in there, no airlock.

**Comment by Steelpoint** [comment]
There's a spot in North East maintenance that might make a good base, but anywhere in maint should do well.

**Comment by Lumipharon** [comment]
Unless it has badmin walls, traitors WILL get in, and proceed to print assloads of ammo for revolvers/stechkins nice and out of sight, just an FYI.

**Comment by Ahammer18** [comment]
That map was just a bit of a demo, someone who can actually map would need to do it.

**Comment by TheVekter** [comment]
Maybe it should have badmin walls then. Or r-walls at least.

**Comment by Ahammer18** [comment]
Oh, and make a special autolathe that only prints tools. Dronelathe.

**Comment by Ahammer18** [comment]
I dont think we can throw around unsimulated turfs without upsetting the balance of the game.

**Comment by Ahammer18** [comment]
Im reworking my design, I'll post once I finish

**Comment by paprka** [comment]
Drones spawn with tools

**Comment by Ahammer18** [comment]
![dronebase 1 1](https://cloud.githubusercontent.com/assets/10344968/6206613/9fa6f1b0-b553-11e4-94c3-5ffe6055ece1.png)


Here, the two fabricators indicate where pap's doublewide sprite would go, as I didn't have it in the outdated version of the code I have.

Got rid of autolathe, replaced with several normal, low security vending machines. Added disposals pipe, removed doors. Fixed pipes.

**Comment by TheVekter** [comment]
I'm working on my version right now. I''ll post it in a few.

**Comment by Ahammer18** [comment]
![dronebase 1 2](https://cloud.githubusercontent.com/assets/10344968/6206953/832c046e-b55b-11e4-80a5-e3d3888cfc9b.png)

Here we are, Added the right pipe colors, fixed disposals pipes, added a better dronefab replacement, added metal and glass.

**Comment by Ahammer18** [comment]
Also keep in mind, this is visual only, none of the stuff is configured correctly and this isnt really a working build to my knowledge. Like I said, I am fairly inexperienced with mapping (this is my first map edit).

**Comment by Niknakflak** [comment]
That's too large and has more tools than engineering.  It should be the size of art storage or something if this is even considered.

**Comment by TheVekter** [comment]
![](http://i.imgur.com/tJqZv15.png)

This is what I was thinking for something a bit smaller. I just had the tool vendors start with metal and glass in them (though I can't for the life of me get the game to recognize plasteel stacks...)

The exofab will be replaced by the dronefab. I can make this another square wider if its necessary.

I know there's no air alarm, I don't have anything wired up properly yet.

**Comment by paprka** [comment]
make it 1 tile wider yes

**Comment by TheVekter** [comment]
![](http://i.imgur.com/LxlTTra.png)
I removed the lathe at the urging of about 5 different people (the concern being that if someone breaks in they get a free autolathe in maintenance). I was considering making a snowflake drone-only lathe but I wanted your guys' input before I start coding it.

Other ideas I heard: Put it on the AI sat, or make its own little sat-room off in space.

**Comment by paprka** [comment]
drone only lathe is shit, stealing from cargo is drone culture

**Comment by Steelpoint** [comment]
Yeah, some spare bags on the side would be extremely useful.

**Comment by TheVekter** [comment]
Bags! Yes, thank you.

Should drones eventually have inventory space?

**Comment by TheVekter** [comment]
![](http://i.imgur.com/l22N9Lp.png)
Finished~

**Comment by Steelpoint** [comment]
Looking good.

**Comment by Lumipharon** [comment]
Why do drones need a tool vendor? They literally spawn with a full toolbox.
Why do drones need a special base? They're literally just an optional thing that dead people can sometimes do.

Why do we want a drone fab (having it as an admin thing is fine), when it's part of robotics being able to make them? Might as well have a golem fab as well.

**Comment by Chocobro** [comment]
research needs to be done before robotics can make them. sort of against this as it removes an area people use for creativity.

**Comment by TheVekter** [comment]
I put the tool vendors in to replace a drone's tools in case he drops them and someone steals 'em. Also it's the easiest way to store metal/glass/etc without clutter. I can replace them with a rack for the metal and such if you think that's better.

The machines also give out hats if they're hacked.

I don't think they NEED a special base, but I don't see any harm in them having somewhere they can come back to and not have to worry about someone picking them up and using them as a hat if they need to go braindead.

The dronafab idea was because there's too many rounds where R&D doesn't get done and hence drones don't get built. This way it's not dependent on research, and players can still get back into the game.

@Chocobro I've literally never seen anyone use that small 8x4 square for anything creative, and considering that entire corner of the station is unused bullshit, I think the crew will manage.

**Comment by Miauw62** [comment]
meh, please make an idea thread.
this is not about something that really should be there and cause issues through their absence like most things under feature request, so i'm just going to close it.
